pkg/logql/expr.y

```diff
@@ -226,6 +233,9 @@ labelParser:
   | REGEXP STRING  { $$ = newLabelParserExpr(OpParserTypeRegexp, $2) }
   ;
 
+jsonExpressionParser:
```

Comment:
you couldn't make that part of `labelParser`

Reply:
I need to return a different result: `newJSONExpressionParser` not `newLabelParserExpr` - and catering for these two different types would add complexity; what benefit do you think we could gain by making it part of `labelParser`?

Reply:
Just wondering why !